pkg/plugins/client.go

```diff
@@ -96,17 +107,29 @@ func abort(start time.Time, timeOff time.Duration) bool {
 	return timeOff+time.Since(start) > time.Duration(defaultTimeOut)*time.Second
 }
 
-func configureTCPTransport(tr *http.Transport, proto, addr string) {
-	// Why 32? See https://github.com/docker/docker/pull/8035.
-	timeout := 32 * time.Second
-	if proto == "unix" {
-		// No need for compression in local communications.
-		tr.DisableCompression = true
-		tr.Dial = func(_, _ string) (net.Conn, error) {
-			return net.DialTimeout(proto, addr, timeout)
+func readTLSConfig(config *sockets.TlsConfig) (*tls.Config, error) {
+	certPool := x509.NewCertPool()
+	file, err := ioutil.ReadFile(config.CA)
+	if err != nil {
+		return nil, err
+	}
+	certPool.AppendCertsFromPEM(file)
+
+	tlsConfig := &tls.Config{
```

Comment:
It would be a good place to be opinionated about ciphersuites:
```go
config := tls.Config{
    MinVersion:               tls.VersionTLS10,
    PreferServerCipherSuites: true,
    CipherSuites: []uint16{
        tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
        tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
        tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
        tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
        tls.TLS_RSA_WITH_AES_128_CBC_SHA,
        tls.TLS_RSA_WITH_AES_256_CBC_SHA,},
}
```